### PR TITLE
add tests for genrou serializaton

### LINKED ISSUE #642: De-serialization fails with dynamic models
The de-serialization of a system containing 

- RoundRotorExponential
- RoundRotorQuadratic
- SalientPoleExponential
- SalientPoleQuadratic

fails.

### PR DESCRIPTION
**Pull request by jd-lara**
Requires https://github.com/NREL-SIIP/InfrastructureSystems.jl/pull/173 

closes #642

### CODE REVIEW

File: src/models/supplemental_constructors.jl [hunk 1]
```diff
@@ -112,3 +112,14 @@ function StaticReserve(
         internal,
     )
 end
+
+""" RoundRotorMachine constructor for de-serialization"""
+function RoundRotorMachine(R, Td0_p, Td0_pp, Tq0_p, Tq0_pp, Xd, Xq, Xd_p, Xq_p, Xd_pp, Xl, Se::Array, ext, γ_d1, γ_q1, γ_d2, γ_q2, γ_qd, states, n_states, internal, )
```
Comment (daniel-thom) [inline src/models/supplemental_constructors.jl:117]: Deserialization requires a kwargs constructor which is auto-generated for this struct.  I suspect that we don't fully understand the problem you were hitting.

File: src/models/supplemental_constructors.jl [hunk 2]
```diff
@@ -112,3 +112,14 @@ function StaticReserve(
...
+    RoundRotorMachine(R, Td0_p, Td0_pp, Tq0_p, Tq0_pp, Xd, Xq, Xd_p, Xq_p, Xd_pp, Xl, tuple(Se[1],Se[2]), ext, γ_d1, γ_q1, γ_d2, γ_q2, γ_qd, states, n_states, internal, )
+end
+
+""" SalientPoleMachine constructor for de-serialization"""
+function SalientPoleMachine(R, Td0_p, Td0_pp, Tq0_pp, Xd, Xq, Xd_p, Xd_pp, Xl, Se::Array, ext, γ_d1, γ_q1, γ_d2, states, n_states, internal, )
+    @assert
```
Comment (daniel-thom) [inline src/models/supplemental_constructors.jl:123]: Same as above.  Also, you have a leftover assert.

**Comment by daniel-thom** [comment]
Is something wrong with this PR? 123 commits and 102 changed files?

**Comment by jd-lara** [comment]
@daniel-thom something went wrong when I rebased. No idea what happened. I will close it and open a new one.

**Pull request closed without merging**